src/model.js

```diff
@@ -2808,14 +2808,26 @@ class Model {
           options.updateOnDuplicate = options.updateOnDuplicate.map(attr => model.rawAttributes[attr].field || attr);
           // Get primary keys for postgres to enable updateOnDuplicate
           options.upsertKeys = _.chain(model.primaryKeys).values().map('field').value();
+
+          let onConflictKeys = [];
           if (Object.keys(model.uniqueKeys).length > 0) {
-            options.upsertKeys = _.chain(model.uniqueKeys)
+            onConflictKeys = _.chain(model.uniqueKeys)
```

Comment:
Unrelated to this PR:

Damn I just realized this whole function is almost 400 lines long. I guess at some point we could do some extraction on some of this logic to make this whole thing more manageable right?

Reply:
Yes do you have a chore list? I would like the feature merged in before we do that refactor.

Reply:
I'm not sure, maybe in issues? I wouldn't be able to say since I'm not a regular contributor.